web_widget_one2many_product_picker/static/src/js/widgets/field_one2many_product_picker.js

```diff
@@ -0,0 +1,462 @@
+            this.$("#product_picker_maximize").remove();
+            this.$btnMaximize = $(qweb.render("One2ManyProductPicker.ButtonMaximize"));
+            this.$btnMaximize
+                .appendTo(this.$el)
+                .on("click", this._onClickMaximize.bind(this));
+        },
+
+        /**
+         * Inject the 'maximize' button
+         */
+        _addTotalZone: function () {
+            this.$("#product_picker_total").remove();
+            this.$totalZone = $(qweb.render("One2ManyProductPicker.Total"));
+            this.$totalZone.appendTo(this.$el);
+            this.updateTotalPrice();
+        },
+
+        /**
+         * Obtain the linked records defined in the options.
+         * If merge is true the current records aren't removed.
+         *
+         * @param {Array} domain
+         * @param {Dictionary} options
+         * @param {Boolean} merge
+         */
+        _getSearchRecords: function (domain, options, merge) {
+            var self = this;
+            var arch = this.view.arch;
+            var field_name = this.options.field_map.product;
+            var field_info = this.view.fieldsInfo[arch.tag][field_name];
+            var model = this.view.viewFields[field_info.name].relation;
+
+            // Launch the rpc request and ensures that we wait for the reply
+            // to continue
+            var sdomain = this._getFullSearchDomain(domain);
+            var soptions = options || {};
+            return new Promise(function (resolve) {
+                var limit = soptions.limit || self.options.records_per_page;
+                var offset = soptions.offset || 0;
+                self._rpc({
+                    model: model,
+                    method: "search_read",
+                    fields: self.search_read_fields,
+                    domain: sdomain,
+                    limit: limit,
+                    offset: offset,
+                    orderBy: soptions.order,
+                }).then(function (results) {
+                    if (merge) {
+                        self._searchRecords = _.union(
+                            self._searchRecords || [],
+                            results
+                        );
+                    } else {
+                        self._searchRecords = results;
+                    }
+                    self._searchOffset = offset + limit;
+                    if (self.renderer) {
+                        self.renderer.setSearchState(self._searchRecords);
+                    }
+                    resolve();
+                });
+            });
+        },
+
+        /**
+         * @param {MouseEvent} evt
+         */
+        _onClickSearchGroup: function (evt) {
+            var self = this;
+            var $btn = $(evt.target);
+            var groupIndex = Number($btn.data("group")) || 0;
+            var group = this.searchGroups[groupIndex];
+            this._searchContext.domain = group.domain;
```

Comment:
I find sort options not working. This change worked for me:
```suggestion
            this._searchContext.domain = group.domain;
            this._searchContext.groupBy = group.order;
```